src/Settings.hs

```diff
@@ -98,11 +98,24 @@ programPath context@Context {..} = do
         let installPath = if install then inplaceInstallPath package else path
         return $ installPath -/- programName context <.> exe
 
--- TODO: Set this from command line
```

Comment:
Do we no longer want to longer want to set this flag from command line?

Reply:
We've already done this in #397, do you mean that?

Reply:
I thought we were planning to switch to `stage1Only` build using a command line flag, and this `TODO` was a reminder of that. Maybe I'm wrong and we never planned that, in which case it's fine to remove it.

Reply:
This, plus `crossCompiling`, is actually quite tricky to handle. Make it compiler time only constants that is set through modifying `UserSettings`, we can use it anywhere. But if it must be passed from command line argument, then it must be wrapped, thus can't mixed with pure code freely.

Reply:
Aha, interesting. Why is `crossCompiling` difficult to use if it is not a pure Boolean? At the moment I don't see any places where you couldn't use an `Action Bool` instead.

Reply:
For example, I might want to use crossCompiling to modify `defaultKnownPackages :: [Package]` directly.

You are right in that in the end the build is a monadic action, so ideally refactoring everything to respect `crossCompiling :: Action Bool` is possible. But that might clutter the code too much.

Reply:
Do you really need to modify `defaultKnownPackages`? It's not supposed to be modified, since it captures all known GHC packages statically. Packages that are actually build by Hadrian are captured by expressions like `stage0Packages` which allow `Action Bool` as conditionals.

Reply:
Oh, I see